src/OpenSpaceToolkit/Astrodynamics/Trajectory.cpp

```diff
@@ -129,6 +136,103 @@ Trajectory Trajectory::Position(const physics::coordinate::Position& aPosition)
+    const Real& aGroundSpeed,
+    const Instant& aStartInstant,
+    const Celestial& aCelestial
+)
+{
+    if (!aStartLLA.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Start LLA");
+    }
+
+    if (!anEndLLA.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("End LLA");
+    }
+
+    if (!aGroundSpeed.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Ground speed");
+    }
+
+    if (!aStartInstant.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Start instant");
+    }
+
+    if ((aStartLLA.getAltitude().inMeters() > Real::Epsilon()) || (anEndLLA.getAltitude().inMeters() > Real::Epsilon()))
+    {
+        throw ostk::core::error::RuntimeError("LLA altitude must be zero.");
+    }
+
+    const Length distance =
+        aStartLLA.calculateDistanceTo(anEndLLA, aCelestial.getEquatorialRadius(), aCelestial.getFlattening());
+
+    const Duration duration = Duration::Seconds(distance.inMeters() / aGroundSpeed);
+
+    const Instant endInstant = aStartInstant + duration;
+
+    const Interval interval = Interval::Closed(aStartInstant, endInstant);
+
+    const Array<Instant> instants = interval.generateGrid(Duration::Seconds(1.0));  // TBI: Make a param?
+
+    return GroundStrip(aStartLLA, anEndLLA, instants, aCelestial);
+}
+
+Trajectory Trajectory::GroundStrip(
+    const LLA& aStartLLA, const LLA& anEndLLA, const Array<Instant>& anInstantArray, const Celestial& aCelestial
+)
+{
+    if (!aStartLLA.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Start LLA");
+    }
+
+    if (!anEndLLA.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("End LLA");
+    }
+
+    if (anInstantArray.getSize() < 2)
+    {
+        throw ostk::core::error::RuntimeError("Atleast 2 instants must be provided.");
+    }
+
+    if ((aStartLLA.getAltitude().inMeters() > Real::Epsilon()) || (anEndLLA.getAltitude().inMeters() > Real::Epsilon()))
+    {
+        throw ostk::core::error::RuntimeError("LLA altitude must be zero.");
+    }
+
+    const Velocity velocity = Velocity::MetersPerSecond({0.0, 0.0, 0.0}, Frame::ITRF());
+
+    Array<State> states = Array<State>::Empty();
+
+    const Duration duration = anInstantArray.accessLast() - anInstantArray.accessFirst();
+
+    for (const auto& instant : anInstantArray)
+    {
+        const Real ratio = (instant - anInstantArray.accessFirst()).inSeconds() / duration.inSeconds();
+
+        const LLA intermediateLLA = aStartLLA.calculateIntermediateTo(
+            anEndLLA, ratio, aCelestial.getEquatorialRadius(), aCelestial.getFlattening()
+        );
+
+        const physics::coordinate::Position position = physics::coordinate::Position::Meters(
+            intermediateLLA.toCartesian(aCelestial.getEquatorialRadius(), aCelestial.getFlattening()), Frame::ITRF()
+        );
+
+        const State state = State(instant, position, velocity).inFrame(Frame::GCRF());
```

Comment:
I would not convert to any specific Frame

```suggestion
        const State state = State(instant, position, velocity);
```

But if you do want to specify an "output" frame maybe could be a parameters (with GCRF as default?)